src/Common/src/TypeSystem/Ecma/EcmaMethod.cs

```diff
@@ -415,5 +416,56 @@ public override PInvokeMetadata GetPInvokeMethodMetadata()
 
             return new PInvokeMetadata(moduleName, name, (PInvokeAttributes)import.Attributes);
         }
+
+        public override PInvokeParameterMetadata[] GetPInvokeParameterMetadata()
+        {
+            if (!IsPInvoke)
+                return default(PInvokeParameterMetadata[]);
+
+            MetadataReader metadataReader = MetadataReader;
+            MethodImport import = metadataReader.GetMethodDefinition(_handle).GetImport();
+            string name = metadataReader.GetString(import.Name);
+            
+            // Spot check the enums match
+            Debug.Assert((int)ParameterAttributes.In == (int)PInvokeParameterAttributes.In);
+            Debug.Assert((int)ParameterAttributes.Out == (int)PInvokeParameterAttributes.Out);
+            Debug.Assert((int)ParameterAttributes.Optional == (int)PInvokeParameterAttributes.Optional);
+            Debug.Assert((int)ParameterAttributes.HasDefault == (int)PInvokeParameterAttributes.HasDefault);
+            Debug.Assert((int)ParameterAttributes.HasFieldMarshal == (int)PInvokeParameterAttributes.HasFieldMarshal);
+
+
+            ParameterHandleCollection parameterHandles = metadataReader.GetMethodDefinition(_handle).GetParameters();
+            List<PInvokeParameterMetadata> parameterMetadataList = new List<PInvokeParameterMetadata>();
+            //  The source to IL compiler may chose to not emit metadata for the return type
+            //  if it considers it not necessary. Here we ensure that we have the first entry for return type ofr consistency.
+            if (parameterHandles.Count == 0)
+            {
+                PInvokeParameterMetadata returnTypeMetadata = new PInvokeParameterMetadata(0, PInvokeParameterAttributes.None, null);
+                parameterMetadataList.Add(returnTypeMetadata);
+            }
+            else
+            {
+                int index = 0;
+                foreach (ParameterHandle parameterHandle in parameterHandles)
+                {
+                    Parameter parameter = metadataReader.GetParameter(parameterHandle);
+
+                    if (index == 0 && parameter.SequenceNumber == 1)
+                    {
+                        PInvokeParameterMetadata returnTypeMetadata = new PInvokeParameterMetadata(0, PInvokeParameterAttributes.None, null);
+                        parameterMetadataList.Add(returnTypeMetadata);
+                        index++;
+                    }
+
+                    Debug.Assert(index == parameter.SequenceNumber);
```

Comment:
Thanks for pointing it out. I wasn't aware that there could be gap between the parameters.